apps/src/applab/dataBrowser/DataOverview.jsx

```diff
@@ -125,9 +81,28 @@ const DataOverview = React.createClass({
 });
 
 export default connect(state => ({
-  view: state.data.view
+  view: state.data.view,
+  tableListMap: state.data.tableListMap || {}
 }), dispatch => ({
   onViewChange(view, tableName) {
     dispatch(changeView(view, tableName));
+  },
+  onTableAdd(tableName) {
+    // Add the table name to the model even though it doesn't exist in the database yet,
+    // so that it appears in the overview even if we don't create a record in it.
+    // addTableName() is idempotent, so we won't double-count the table name later
+    // when the the first record is added to it.
+    dispatch(addTableName(tableName));
+
+    dispatch(changeView(DataView.TABLE, tableName));
+  },
+  onTableDelete(tableName) {
+    // Explicitly remove the table from the model to make sure it disappears from the
+    // list, since we won't receive a delete event from the database if the table did
+    // not exist in the database. This could happen when deleting a table after adding it
+    // without adding any records to it.
+    dispatch(deleteTableName(tableName));
+
+    FirebaseStorage.deleteTable(tableName);
```

Comment:
This is a little bit more of an interesting one. I'm not too sure where the right place for `FirebaseStorage.deleteTable is`. I think it probably makes the most sense to treat it like an async web request (even though we don't change our store at all based on the success/failure criteria).

I would create an redux-thunk action that looks something like this:
```
function deleteTable(tableName => {
  return dispatch => {
    dispatch(deleteTableName(tableName));
    FirebaseStorage.deleteTable(tableName));
 };
});
```
(By using redux-thunk, we dont do the delete when we create the action but instead only when it is dispatched).

Then here in `mapDispatchToProps` we would just dispatch(deleteTable);